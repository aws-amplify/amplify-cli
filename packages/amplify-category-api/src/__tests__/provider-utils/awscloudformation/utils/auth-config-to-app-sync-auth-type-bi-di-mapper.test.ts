import {
  authConfigToAppSyncAuthType,
  appSyncAuthTypeToAuthConfig,
} from '../../../../provider-utils/awscloudformation/utils/auth-config-to-app-sync-auth-type-bi-di-mapper';
import {
  AppSyncAPIKeyAuthType,
  AppSyncAWSIAMAuthType,
  AppSyncCognitoUserPoolsAuthType,
  AppSyncOpenIDConnectAuthType,
} from 'amplify-headless-interface';

describe('authConfig to AppSyncAuthType', () => {
  it('maps API_KEY auth correctly', () => {
    const authConfig = {
      authenticationType: 'API_KEY',
      apiKeyConfig: {
        apiKeyExpirationDays: 120,
        description: 'api key description',
      },
    };

    expect(authConfigToAppSyncAuthType(authConfig)).toMatchSnapshot();
  });

  it('maps AWS_IAM auth correctly', () => {
    const authConfig = {
      authenticationType: 'AWS_IAM',
    };

    expect(authConfigToAppSyncAuthType(authConfig)).toMatchSnapshot();
  });

  it('maps AMAZON_COGNITO_USER_POOLS auth correctly', () => {
    const authConfig = {
      authenticationType: 'AMAZON_COGNITO_USER_POOLS',
      userPoolConfig: {
        userPoolId: 'userPoolId',
      },
    };

    expect(authConfigToAppSyncAuthType(authConfig)).toMatchSnapshot();
  });

  it('maps OPENID_CONNECT auth correclty', () => {
    const authConfig = {
      authenticationType: 'OPENID_CONNECT',
      openIDConnectConfig: {
        name: 'openid name',
        issuerUrl: 'issuer url',
        clientId: 'client id',
        authTTL: 'auth TTL',
        iatTTL: 'iat TTL',
      },
    };

    expect(authConfigToAppSyncAuthType(authConfig)).toMatchSnapshot();
  });

  it('returns undefined on undefined input', () => {
    expect(authConfigToAppSyncAuthType(undefined)).toBeUndefined();
  });
});

describe('AppSyncAuthType to authConfig', () => {
  it('maps API_KEY correctly', () => {
    const authType: AppSyncAPIKeyAuthType = {
      mode: 'API_KEY',
      expirationTime: 120,
    };
    expect(appSyncAuthTypeToAuthConfig(authType)).toMatchSnapshot();
  });

  it('maps AWS_IAM correctly', () => {
    const authType: AppSyncAWSIAMAuthType = {
      mode: 'AWS_IAM',
    };
    expect(appSyncAuthTypeToAuthConfig(authType)).toMatchSnapshot();
  });

  it('maps AMAZON_COGNITO_USER_POOLS correctly', () => {
    const authType: AppSyncCognitoUserPoolsAuthType = {
      mode: 'AMAZON_COGNITO_USER_POOLS',
      cognitoUserPoolId: 'someID',
    };
    expect(appSyncAuthTypeToAuthConfig(authType)).toMatchSnapshot();
  });

  it('maps OPENID_CONNECT correctly', () => {
    const authType: AppSyncOpenIDConnectAuthType = {
      mode: 'OPENID_CONNECT',
      openIDProviderName: 'providerName',
      openIDClientID: 'client id',
      openIDIssuerURL: 'issuer url',
    };

    expect(appSyncAuthTypeToAuthConfig(authType)).toMatchSnapshot();
  });

  it('returns undefined on undefined input', () => {
    expect(appSyncAuthTypeToAuthConfig(undefined)).toBeUndefined();
  });
});
