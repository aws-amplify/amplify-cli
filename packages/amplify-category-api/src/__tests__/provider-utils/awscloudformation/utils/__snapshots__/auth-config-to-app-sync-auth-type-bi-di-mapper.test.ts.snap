// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSyncAuthType to authConfig maps AMAZON_COGNITO_USER_POOLS correctly 1`] = `
Object {
  "authenticationType": "AMAZON_COGNITO_USER_POOLS",
  "userPoolConfig": Object {
    "userPoolId": "someID",
  },
}
`;

exports[`AppSyncAuthType to authConfig maps API_KEY correctly 1`] = `
Object {
  "apiKeyConfig": Object {
    "apiKeyExpirationDate": undefined,
    "apiKeyExpirationDays": 120,
    "description": undefined,
  },
  "authenticationType": "API_KEY",
}
`;

exports[`AppSyncAuthType to authConfig maps AWS_IAM correctly 1`] = `
Object {
  "authenticationType": "AWS_IAM",
}
`;

exports[`AppSyncAuthType to authConfig maps OPENID_CONNECT correctly 1`] = `
Object {
  "authenticationType": "OPENID_CONNECT",
  "openIDConnectConfig": Object {
    "authTTL": undefined,
    "clientId": "client id",
    "iatTTL": undefined,
    "issuerUrl": "issuer url",
    "name": "providerName",
  },
}
`;

exports[`authConfig to AppSyncAuthType maps AMAZON_COGNITO_USER_POOLS auth correctly 1`] = `
Object {
  "cognitoUserPoolId": "userPoolId",
  "mode": "AMAZON_COGNITO_USER_POOLS",
}
`;

exports[`authConfig to AppSyncAuthType maps API_KEY auth correctly 1`] = `
Object {
  "apiKeyExpirationDate": undefined,
  "expirationTime": 120,
  "keyDescription": "api key description",
  "mode": "API_KEY",
}
`;

exports[`authConfig to AppSyncAuthType maps AWS_IAM auth correctly 1`] = `
Object {
  "mode": "AWS_IAM",
}
`;

exports[`authConfig to AppSyncAuthType maps OPENID_CONNECT auth correclty 1`] = `
Object {
  "mode": "OPENID_CONNECT",
  "openIDAuthTTL": "auth TTL",
  "openIDClientID": "client id",
  "openIDIatTTL": "iat TTL",
  "openIDIssuerURL": "issuer url",
  "openIDProviderName": "openid name",
}
`;
